File: Deliverables/BizBong/app/src/main/java/gamesoftitalia/bizbong/entity/Domanda.java
package gamesoftitalia.bizbong.entity;

import gamesoftitalia.bizbong.entity_interface.DomandaInterface;

import java.io.Serializable;
import java.util.List;

/**
 * Created by GameSoftItalia on 01/12/2016.
 */
public class Domanda implements DomandaInterface, Serializable{

    private int punteggio;
    private String tema, domanda, rispostaVera;
    private List<String> listaRisposte;

    public Domanda(int punteggio, String tema, String domanda, String rispostaVera, List<String> listaRisposte) {
        this.punteggio = punteggio;
        this.tema = tema;
        this.domanda = domanda;
        this.rispostaVera = rispostaVera;
        this.listaRisposte = listaRisposte;
    }

    @Override
    public int getPunteggio() {
        return punteggio;
    }

    @Override
    public String getTema() {
        return tema;
    }

    @Override
    public String getDomanda() {
        return domanda;
    }

    @Override
    public List getListaRisposte() {
        return listaRisposte;
    }

    @Override
    public String getRispostaVera() {
        return rispostaVera;
    }

    @Override
    public void setPunteggio(int punteggio) {
        this.punteggio = punteggio;
    }

    @Override
    public void setTema(String tema) {
        this.tema = tema;
    }

    @Override
    public void setDomanda(String domanda) {
        this.domanda = domanda;
    }

    @Override
    public void setListaRisposte(List<String> listaRisposte) {
        this.listaRisposte = listaRisposte;
    }

    @Override
    public void setRispostaVera(String rispostaVera) {
        this.rispostaVera = rispostaVera;
    }

    @Override
    public String toString(){
        String risposte = "{";
        for(int i = 0; i < listaRisposte.size(); i++){
            risposte += listaRisposte.get(i)+";";
        }
        return "Domande[punteggio:"+punteggio+", tema:"+tema+", domanda:"+domanda+", listaRisposte:"+risposte+"}, rispostaVera:"+rispostaVera+"]";
    }

    @Override
    public boolean equals(Object obj){
        if (obj == null) return false;

        if (!(obj instanceof Domanda)) return false;

        Domanda d = (Domanda) obj;
        return ( this.punteggio == d.punteggio && this.tema.equals(d.tema) && this.domanda.equals(d.domanda)
                && this.listaRisposte.equals(d.listaRisposte) && this.rispostaVera.equals(d.rispostaVera));
    }
}

File: Deliverables/php/Entity/Domanda.php
<?php
	//**********************************************************************************************
    // Creazione dell' oggetto Domanda che implementa JsonSerializable per essere inviato tramite json
    // Variabili: punteggio, tema, domanda, listaDomande, rispostaVera
    // jsonSerialize() --> utilizzato per la conversione a json
    //**********************************************************************************************
	
    //include ".../Libreria/ArrayList.php";
    
    class Domanda implements JsonSerializable {
    	private $punteggio;
        private $tema;
        private $domanda;
        private $listaRisposte;
        private $rispostaVera;
        
        public function __construct($punteggio, $tema, $domanda, $listaRisposte, $rispostaVera){
            $this->punteggio = $punteggio;
            $this->tema = $tema;
            $this->domanda = $domanda;
            $this->listaRisposte = $listaRisposte;
            $this->rispostaVera = $rispostaVera;
        }
        
        public function getPunteggio(){
        	return $this->punteggio;
        }
        
        public function getTema(){
        	return $this->tema;
        }
        
        public function getDomanda(){
        	return $this->domanda;
        }
        
        public function getListaRisposte(){
        	return $this->listaRisposte;
        }
        
        public function getRispostaVera(){
        	return $this->rispostaVera;
        }
        
        public function setPunteggio($punteggio){
        	$this->punteggio = $punteggio;
        }
        
        public function setTema($tema){
        	$this->tema = $tema;
        }
        
        public function setDomanda($domanda){
        	$this->domanda = $domanda;
        }
        
        public function setListaRisposte($listaRisposte){
        	$this->listaRisposte = $listaRisposte;
        }
        
        public function setRispostaVera($rispostaVera){
        	$this->rispostaVera = $rispostaVera;
        }
              
        public function jsonSerialize() {
        	return get_object_vars($this);
    	}
    }
?>